### ISSUE TITLE
Watch charging status and battery level not updating since rc9?

### ISSUE THREAD

**Issue by jayscovill** [gcobb321/icloud3_v3#249]
It seems that since the rc9 update the new icloud3.action service doesn't update the charge status and battery level of my Watch?

Initially when I configure my watch for location updates from Family Sharing and using the iOS App it updates the battery level and charging status.

There were times I wanted to update those entities faster than the default 30 minutes I used to be able to call the iclould3.update service and it would update those entities.  Now when I use the new icloud3.action service, selecting either family sharing or "Send Locate Request to iOS App"  and my watch, it doesn't update them, only location.

**Comment by jayscovill** [comment]
Does the icloud3 integation even track Watch battery level and charging status anymore?  It definitely used to as it was using it in an automation to notify me if I was leaving the house during a specific time of the say and the watch status was charging.

**Comment by gcobb321** [comment]
The watch is tracked by iCloud3 using the FamShr tracking method. It is not tracked with the iOSApp. There is an HA Complication but it is limited to actionable notifications only, the complication does not update any of the iOS APP (Mobile App Integration) sensors as far as I can tell.

My wife has a Watch. I just put it on the charger and did an _Event Log > Actions > Locate using FamShr_. It woke up, provided the location information and the battery level/status was updated as expected from Not Charging to Charging. See the screenshots below.

The Watch is set up like any other device. Have you recently upgraded to a new watch? If so, did you check/update the iCloud3 Devices screen for the new watch's Apple ID info?
![watch-evlog](https://github.com/gcobb321/icloud3_v3/assets/35449479/fe7cbc8d-b36e-4236-86b6-136e7b095df1)
![watch-battery-state](https://github.com/gcobb321/icloud3_v3/assets/35449479/f801a7b6-d592-4eac-975f-30ce9dd58df4)

**Comment by jayscovill** [comment]
I did reset my watch but I've since removed and re-added it to the integration as well as completely removed the integration and then set it up again and it still didn't update the battery charge or status.

When I set it up again it didn't require me to re-auth to Apple and it still found both my phone and watch so obviously the creds are cached but there may be bad info cached as well that I need to get rid of.

**Comment by gcobb321** [comment]
1. Authentication and session cookies are stored in the _config/.storage/icloud_ directory. They are token data only that provides access to the account. No account data is stored in them. 
2. iCloud3 stores its config in the _config/.storage/icloud3/configuration_ file. It is not deleted when the iCloud3 integration is removed so deleting and reinstalling doesn’t reinitialize the config. Deleting the device on the _Configure Settings > icloud3 Devices_ screen and saving the change will delete it. 
3. When iCloud3 starts, the iCloud account is logged into and a list of the devices it’s received from Apple. Each device is identified by a device name from the _Settings App > General >About > Name_ from the device and an internal 25+ character device_id value. The device is is what identifies each device and is assigned by Apple when the device is added to your account Family Sharing List. I do not know if it it’s assigned a new number or reuses its old number if it is deleted and readd to the list. I would suspect it is reused and its value is hardware based. 
4. The tracking info and battery data is associated with the device using the device ID. That value is stored in the iCloud3 config file. When you change devices and set it up, a suffix is added to the name (2) of the new device and both devices are returned during startup until it is removed from the Family Sharing list. If the device name is found during startup, the device id is checked and updated in the iCloud3 config file and it it’s tracked. 
5. Do this:
- Review the devices on the FindMy app and make sure there are correct and tracked and current. 
- Make sure Location Sharing is on. 
- Review the devices returned from Apple during startup in iCloud3 Event Log Stage 4. 
- Select _Event Log > Actions > Show Tracking Monitors_ to see more detail on the startup data and when a device is located. Look at the u Update times, Device ID, device model, etc and make sure the correct device is being used. 
- Select _Event Log > Actions > Start RawData  Logging_ (or something like that) and restart HA. That dumps everything to the _icloud3-0.log_ file. Email that to geekstergary@gmail.com and I’ll see if I see anything.

**Comment by jayscovill** [comment]
Thanks for the details.  I set up my phone just fine and it's updating.  But when I go to add my watch I get "Login Error, iCloud Account is not Available" at the screen after the initial add device setup.
<img width="572" alt="Screenshot 2024-01-03 at 10 47 44 AM" src="https://github.com/gcobb321/icloud3_v3/assets/30603659/115cda1e-e0ad-4d3c-a846-255c249a1524">

**Comment by jayscovill** [comment]
Well, managed to get around that message but now the iCloud integration isn't seeing my watch in the list of devices...Confirmed that it's shows up in the FindMy app.

**Comment by gcobb321** [comment]
I'll look into that. Meanwhile, exit the Configure Settings screens and then go back in to add the watch. 

Did you go into the Configure Settings screen, add the iCloud username/password, go to Add Devices and add the iPhone, and that was all OK. Then you got the eror adding the Watch?

**Comment by jayscovill** [comment]
Yeah, was able to add my phone ok.  No more error when trying to add my watch but it just doesn't show up in the devices dropdown.  All my other devices available through Family Sharing/FmF are there.  Just not my watch for some reason.

**Comment by gcobb321** [comment]
Look at Event Log > Stage 4 when iC3 starts. Are all of the device in the FindMy listed there? If not, check the FindMy and make sure the missing device has location sharing enabled. If they are listed, Restart iC3 on the Event Log > Actions, then go back into the Update Devices screen again. The list is rebuilt each time iC3 starts

**Comment by gcobb321** [comment]
Sharing your location has has always been the case to see devices other than the one under the Me selection on the FibdMy App. iCloud3 needs to be restarted when you make device changes on the app to pick up the changes.

Gary Cobb
________________________________

**Comment by jayscovill** [comment]
Wow...ok, a) I swear I've never had that on before and b) I also swear that I tried enabling it and restarting and it still didn't show up.  But now after enabling and restarting it looks like it's good to go. Mind you I've also reinstalled the integration several times since so maybe that fixed something else that was wrong other than me just being dense.

Really sorry this was so much work for such a small Layer 8 error.  I REALLY appreciate all the effort you put into helping me out with this!
